src/components/app/userActionRowCTA/constants.tsx

```
@@ -94,3 +95,20 @@ export const USER_ACTION_ROW_CTA_INFO: Record<
     WrapperComponent: UserActionFormNFTMintDialog,
   },
 }
+
+export const USER_ACTION_ROW_CTA_INFO_FROM_CAMPAIGN: Record<
+  string,
+  Omit<UserActionRowCTAProps, 'state'>
+> = {} // This is a temp placeholder. This will be updated with the actual CNN campaign data
+
+export function getUserActionCTAInfo(actionType: ActiveClientUserActionType, campaign?: string) {
+  if (!campaign || campaign === USER_ACTION_TO_CAMPAIGN_NAME_DEFAULT_MAP[actionType]) {
+    return USER_ACTION_ROW_CTA_INFO[actionType]
+  }
+
+  if (USER_ACTION_ROW_CTA_INFO_FROM_CAMPAIGN[campaign]) {
+    return USER_ACTION_ROW_CTA_INFO_FROM_CAMPAIGN[campaign]
+  }
+
+  return null
+}
```

Wouldn't it be better to return `USER_ACTION_ROW_CTA_INFO[actionType]` instead of `null` in case the campaign doesn't have a cta info? Would simplify usage by not having to deal with null cases

---

My idea on returning null is to prevent any bugs if a dev forget to change on necessary files. For example: If you add  a new CTA to the priority array or to the additional campaigns, but you don't set it here in this file, it won't show anything. If I just returned the default one, we could have scenarios of duplicate CTAs because of this. But I can change this, np. Just tried to prevent unexpected scenarios

---

In other places we're relying on type safety for that, I think we should use that approach here too, that way we enforce the engineer doesn't forget anything while not introducing unexpected behaviors in the app.

And actually if that's never expected we could actually throw an error instead of proceeding here

---

Thanks for the suggestion! Updated the PR